admission-policy/readme.adoc

```diff
@@ -0,0 +1,405 @@
+itself is a general-purpose policy engine and knows nothing about Kubernetes.
+kube-mgmt is a collection of Kubernetes-specific code that helps
+OPA interact with kubernetes.
+
+**opa-admission-controller-deployment.yaml**:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    app: opa
+  name: opa
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: opa
+      name: opa
+    spec:
+      containers:
+        - name: opa
+          image: openpolicyagent/opa:0.5.13
+          args:
+            - "run"
+            - "--server"
+            - "--tls-cert-file=/certs/tls.crt"
+            - "--tls-private-key-file=/certs/tls.key"
+            - "--addr=0.0.0.0:443"
+            - "--insecure-addr=127.0.0.1:8181"
+          volumeMounts:
+            - readOnly: true
+              mountPath: /certs
+              name: opa-server
+        - name: kube-mgmt
+          image: openpolicyagent/kube-mgmt:0.4
+          args:
+            - "--replicate=v1/pods"
+            - "--register-admission-controller"
+            - "--admission-controller-ca-cert-file=/certs/ca.crt"
+            - "--admission-controller-service-name=opa"
+            - "--admission-controller-service-namespace=$(MY_POD_NAMESPACE)"
+          volumeMounts:
+            - readOnly: true
+              mountPath: /certs
+              name: opa-ca
+          env:
+            - name: MY_POD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+      volumes:
+        - name: opa-server
+          secret:
+            secretName: opa-server
+        - name: opa-ca
+          secret:
+            secretName: opa-ca
+```
+
+```bash
+kubectl create -f opa-admission-controller-deployment.yaml -n opa
+```
+
+When OPA starts, the sidecar (`kube-mgmt`) will register it as an External
+Admission Controller. To verify that registration succeeded, query
+ the Kubernetes API for the list of External Admission Controllers.
+
+```bash
+kubectl describe externaladmissionhookconfigurations admission.openpolicyagent.org
+```
+
+
+
+Finally, you can follow the OPA logs to see the webhook requests being issued
+by the Kubernetes API server:
+
+```
+kubectl logs -l app=opa -c opa -n opa
+```
+
+== Step 3:  Enforce Kubernetes Admission Control with OPA
+
+=== 3.1 Load a policy into OPA
+To test admission control, create a policy that requires all images
+to come from an AWS repository.  For details on the policy language, see the
+http://www.openpolicyagent.org/docs/[Open Policy Agent] documentation.
+
+**image_source.rego**:
+
+```ruby
+package system
+
+# Deny requests that include container images not from ECR.
+deny[explanation] {
+    image_name = input.spec.object.Spec.Containers[_].Image
+    image_name_parts = split(image_name, "/")
+    repo_name = image_name_parts[0]
+    not startswith(repo_name, "12345678.dkr.ecr.us-west-2.amazonaws.com")
```

Comment:
Should we add a note above this file like we do with the pod manifest about changing the account id?

Reply:
Good catch!  Fixed in next patch.